prefix-hex/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! The `prefix-hex` crates offers encoding and decoding of hex strings with a `0x` prefix.
+//!
+//! Its API aims to mimic that of the [`hex`](https://docs.rs/hex/latest/hex/) crate, which we also use internally.
+//!
+//! This crate is compatible with the hex encoding rules of the [Ethereum RPC API](https://eth.wiki/json-rpc/API#hex-value-encoding).
+
+#![no_std]
+
+extern crate alloc;
+
+mod data;
+mod error;
+#[cfg(feature = "primitive-types1")]
+mod primitive_types;
+
+use alloc::string::String;
+
+pub use error::Error;
+
+/// Tries to decode an hexadecimal encoded string with a `0x` prefix.
+pub trait FromHexPrefixed: Sized {
+    /// Tries to decode an hexadecimal encoded string with a `0x` prefix.
+    fn from_hex_prefix(hex: &str) -> Result<Self, Error>;
+}
+
+// TODO: Maybe introduce `handle_error` function with `#[cold]` attribute.
+fn strip_prefix(hex: &str) -> Result<&str, Error> {
```

Comment:
Bit OCD but can you put it after the traits at the end ?

Reply:
Also, why can this be used if it's private 🤔 ?

Reply:
This is in `lib.rs`, so it can still be accessed in descendent modules, right? Just not outside of the crate. I think that's how it works here anyway 😅

Reply:
I didn't know that, thanks :)